### PR TITLE
VZ-10840 ensure module catalog is up to date

### PR DESCRIPTION
**Pull request by mbenmclean**
This change runs unit tests against the module catalog to ensure that it is up to date with the component list and the bom.

### CODE REVIEW

File: platform-operator/verrazzano-bom.json [hunk 1]
```diff
@@ -728,7 +755,8 @@
       ]
     },
     {
-      "name": "node-exporter",
+      "name": "prometheus-node-exporter",
```
Comment (michael-cico) [inline platform-operator/verrazzano-bom.json:758]: same question, changing this key we should make sure lines up with any references in the component code.

File: platform-operator/verrazzano-bom.json [hunk 2]
```diff
@@ -745,7 +773,7 @@
       ]
     },
     {
-      "name": "jaeger",
+      "name": "jaeger-operator",
```
[the remark this answers is not in the document]
Comment (mbenmclean) [reply platform-operator/verrazzano-bom.json:776]: Not that I could find.

File: platform-operator/verrazzano-bom.json (hunk 1 above)
Comment (mbenmclean) [reply platform-operator/verrazzano-bom.json:758]: Not that i could find

File: platform-operator/verrazzano-bom.json [hunk 3]
```diff
@@ -19,8 +18,8 @@
       ]
     },
     {
-      "name": "ingress-nginx",
-      "version": "v1.7.1",
+      "name": "ingress-controller",
```
[the remark this answers is not in the document]
Comment (mbenmclean) [reply platform-operator/verrazzano-bom.json:21]: Not that I could find

**Review by pfmackin — APPROVED** [review]

File: platform-operator/experimental/catalog/catalog.yaml [hunk 4]
```diff
@@ -37,15 +37,15 @@ modules:
   - name: verrazzano-application-operator
```
Comment (michael-cico) [inline platform-operator/experimental/catalog/catalog.yaml:37]: we're going to track VAO, etc with the VZ version via some kind of template scheme in a follow-up right?

File: platform-operator/experimental/catalog/catalog_test.go [hunk 5]
```diff
@@ -5,14 +5,75 @@ package catalog
...
+var subcomponentOverrides = map[string]bomSubcomponentOverrides{
+	"cluster-api":                 {subcomponentName: "capi-cluster-api", imageName: "cluster-api-controller"},
+	"fluentbit-opensearch-output": {subcomponentName: "fluent-operator", imageName: "fluent-bit"},
+	"opensearch":                  {subcomponentName: "verrazzano-monitoring-operator", imageName: "opensearch"},
+	"opensearch-dashboards":       {subcomponentName: "verrazzano-monitoring-operator", imageName: "opensearch-dashboards"},
+	"grafana":                     {subcomponentName: "verrazzano-monitoring-operator", imageName: "grafana"},
+}
+
+// TestCatalogModuleVersions makes sure the module versions in the catalog are up-to-date with the Bom
+func TestNewCatalogModuleVersions(t *testing.T) {
```
Comment (michael-cico) [inline platform-operator/experimental/catalog/catalog_test.go:38]: At present, this won't run I think while it's in the experimental dir.  We may need to update the build to run these.

**Review by michael-cico — APPROVED** [review]

File: platform-operator/experimental/catalog/catalog.yaml (hunk 4 above)
Comment (mbenmclean) [reply platform-operator/experimental/catalog/catalog.yaml:37]: yes. That's my followup commit

File: platform-operator/experimental/catalog/catalog_test.go (hunk 5 above)
Comment (mbenmclean) [reply platform-operator/experimental/catalog/catalog_test.go:38]: Oh I didn't realize that. I'll add that back in in my followup commit

**Review by michael-cico — APPROVED** [review]

**Pull request merged**